File: procon-api/src/main/java/br/com/procon/models/auxiliares/Endereco.java
package br.com.procon.models.auxiliares;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;

import br.com.procon.models.enums.UF;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 
 * @author Marlon F. Garcia
 *
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@Embeddable
public class Endereco implements Serializable {

	private static final long serialVersionUID = -1215304834350865024L;
	@Column(length = 30)
	private String cep;
	private String logradouro;
	@Column(length = 30)
	private String numero;
	@Column(length = 50)
	private String complemento;
	private String bairro;
	private String municipio;
	@Enumerated(EnumType.STRING)
	@Column(length = 2)
	private UF uf;

}

File: procon-front/src/app/models/auxiliares/endereco.ts
export class Endereco {
  cep: string;
  logradouro: string;
  numero: string;
  complemento: string;
  bairro: string;
  municipio: string;
  uf: string;

  constructor();
  constructor(cep: string, logradouro: string, numero: string, complemento: string,
    bairro: string, municipio: string, uf: string);
  constructor(...args: any[]) {
    if (args.length == 0) {
      this.cep = '';
      this.logradouro = '';
      this.numero = '';
      this.complemento = '';
      this.bairro = '';
      this.municipio = '';
      this.uf = '';
    } else {
      this.cep = args[0];
      this.logradouro = args[1];
      this.numero = args[2];
      this.complemento = args[3];
      this.bairro = args[4];
      this.municipio = args[5];
      this.uf = args[6];
    }
  }
}
